### PR TITLE
Add integration test and widget test flow to playground example

### PR DESCRIPTION
**Pull request by AcarFurkan**
## Description

I added integration and widget tests to the playground example to see whether the integrity is preserved when changes are made to the code.

https://github.com/woltapp/wolt_modal_sheet/assets/65075121/2b250ff6-cff7-4bba-87f7-493f3ac4b16e

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors.
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] The package compiles with the minimum Flutter version stated in the [pubspec.yaml](https://github.com/woltapp/wolt_modal_sheet/blob/main/pubspec.yaml#L8)

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[Contributor Guide]: https://github.com/woltapp/wolt_modal_sheet/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by AcarFurkan** [comment]
image
 failed because dart version was old. what should l do?

File: playground/integration_test/playground_flow_test.dart [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    ($) async {
+      await $.pumpWidgetAndSettle(
+        const MaterialApp(
+          home: DemoApp(
+            androidSdkVersion: 30,
+          ),
+        ),
+      );
+      //Page with forced max height and background color flow
+      expect($('Wolt Modal Sheet Playground'), findsOneWidget);
```
Comment (yunusemrebakir) [inline playground/integration_test/playground_flow_test.dart:19]: Thanks Furkan for the PR! We aim to not use text as finders as they likely to change more often compared to other finders. Could you please update to use other finders like; keys, type, widget, etc.

File: playground/integration_test/playground_flow_test.dart [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import 'package:demo_ui_components/demo_ui_components.dart';
+import 'package:flutter/material.dart';
+import 'package:patrol/patrol.dart';
+import 'package:playground/main.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  patrolTest(
```
Comment (yunusemrebakir) [inline playground/integration_test/playground_flow_test.dart:8]: We aim to minimize the use of third-party libraries in our project to make it easier for new contributors to join. Can you please re-write without patrol.
We will provide a test template soon that explains our approach and style.

**Review by yunusemrebakir — CHANGES_REQUESTED** [review]

File: playground/integration_test/playground_flow_test.dart (hunk 1 above)
Comment (AcarFurkan) [reply playground/integration_test/playground_flow_test.dart:19]: Yes, you are absolutely right, I made it a little bit like a demo, from now on I will only use types and keys.

File: playground/integration_test/playground_flow_test.dart (hunk 2 above)
Comment (AcarFurkan) [reply playground/integration_test/playground_flow_test.dart:8]: Ah thank you i will handle with core test libs

**Comment by yunusemrebakir** [comment]
Hey @AcarFurkan we are doing changes and adding more guidelines to the repo. We will soon add a test case sample as a guideline. To unify how we write tests, I will close this PR for now. After the guideline, please feel free to add more tests, we need them 🙌🏽

**Pull request closed without merging**